### PR TITLE
Orb poll hrt

### PR DESCRIPTION
**Pull request by jlaitine**
Replace orb polling with much simpler way to trigger periodic updates with HRT and wakeup the mavlink using HRT timer.

### CODE REVIEW

File: src/modules/mavlink/mavlink_receiver.cpp [hunk 1]
```diff
@@ -2338,18 +2337,59 @@ MavlinkReceiver::handle_message_hil_sensor(mavlink_message_t *msg)
...
-	}
-
 	mavlink_hil_gps_t hil_gps;
 	mavlink_msg_hil_gps_decode(msg, &hil_gps);
 
-	_hitl_sim_gps_time_usec = hil_gps.time_usec;
-	_hitl_sim_home_lat = hil_gps.lat;
-	_hitl_sim_home_lon = hil_gps.lon;
-	_hitl_sim_home_alt = hil_gps.alt;
+	sensor_gps_s gps{};
```
Comment (pussuw) [inline src/modules/mavlink/mavlink_receiver.cpp:2343]: Looks like this is unrelated ?

File: src/modules/mavlink/mavlink_params.c [hunk 2]
```diff
@@ -156,15 +156,6 @@ PARAM_DEFINE_INT32(MAV_HB_FORW_EN, 1);
  */
 PARAM_DEFINE_INT32(MAV_RADIO_TOUT, 5);
 
-/**
```
Comment (pussuw) [inline src/modules/mavlink/mavlink_params.c:159]: Are these HITL changes related to this topic or ?

**Review by pussuw — APPROVED** [review]
LGTM, seems like there is some unrelated HITL stuff here as well, fine by me tbh

File: src/modules/mavlink/mavlink_receiver.cpp (hunk 1 above)
Comment (jlaitine) [reply src/modules/mavlink/mavlink_receiver.cpp:2343]: Yes, I had temporarily reverted one patch which breaks the sitl. Will remove that!

File: src/modules/mavlink/mavlink_params.c (hunk 2 above)
Comment (jlaitine) [reply src/modules/mavlink/mavlink_params.c:159]: No, it was a revert to be able to run this in sitl for now

**Comment by jlaitine** [comment]
> LGTM, seems like there is some unrelated HITL stuff here as well, fine by me tbh

Thanks for checking it out! This is still WIP, there is some instability when stopping mavlink. I will also change at least the following:

- change timestamp in uorb device nodes into px4_atomic to support 32-bit targets
- instead of launching one hrt timer for each stream, make logic to find "root timers"  i.e. common denominator intervals and only run hrt for those

This is now an initial version to check the performance (which improves already significantly). When complete, I will mark it as ready for review, but any comments are of course welcome!

**Comment by jlaitine** [comment]
I still removed the orb polling change from this PR; it is no longer related and better not to introduce other changes than this mavlink poll

**Comment by jlaitine** [comment]
Now tested on desk and ready. A bit hard to review as most of stuff is reverts, better to just check the actual implementation commits

**Comment by jlaitine** [comment]
I just realised a bug in the code; the interval of -1 (unlimited) is not handled. I need to revisit the code to manage that. Also there are couple of unused functions.
